certification/internal/policy/operator/scc_check.go

```diff
@@ -0,0 +1,121 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/bundle"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+)
+
+// SecurityContextConstraintsCheck evaluates the csv and logs a message if a non default security context constraint is
+// needed by the operator
+
+type SecurityContextConstraintsCheck struct {
+	customSCCSpecified  bool
+	builtInSCCSpecified bool
+}
+
+func getDefaultSCCs() []string {
+	return []string{"restricted", "privileged", "nonroot", "node-exporter", "hostnetwork", "hostmount-anyuid", "hostaccess", "anyuid"}
+}
+
+func NewSecurityContextConstraintsCheck() *SecurityContextConstraintsCheck {
+	return &SecurityContextConstraintsCheck{
+		false,
+		false,
+	}
+}
+
+func (p *SecurityContextConstraintsCheck) Validate(ctx context.Context, bundleRef certification.ImageReference) (bool, error) {
+	requestedSCCList, err := p.dataToValidate(ctx, bundleRef.ImageFSPath)
+	if err != nil {
+		return false, err
+	}
+
+	return p.validate(ctx, requestedSCCList)
+}
+
+func (p *SecurityContextConstraintsCheck) dataToValidate(ctx context.Context, imagePath string) ([]string, error) {
+	csvFilepath, err := bundle.GetCsvFilePathFromBundle(imagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	csvFileReader, err := os.Open(csvFilepath)
+	if err != nil {
+		return nil, err
+	}
+
+	requestedSccList, err := bundle.GetSecurityContextConstraints(ctx, csvFileReader)
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract security context constraints from ClusterServiceVersion: %w", err)
+	}
+
+	return requestedSccList, nil
+}
+
+func (p *SecurityContextConstraintsCheck) validate(ctx context.Context, requestedSccList []string) (bool, error) {
+	defaultSccList := getDefaultSCCs()
+
+	if requestedSccList == nil || len(requestedSccList) == 0 {
+		log.Infof("No security context constraint was requested so using the default %s scc", defaultSccList[0])
```

Comment:
I would argue that we don't really need to log this, right? We're just concerned if the developer plans on using _custom_ SCCs?

Reply:
part of the ask 🤷 "If nothing is specified in the CSV, preflight should indicate that the default will be used."

Reply:
Gotcha. I'm not sure what it adds, but if it's a part of the ask I suppose it can stay. That said, can we reword?

```suggestion
		log.Infof("No custom security context constraint was detected in this bundle so using the default %s scc", defaultSccList[0])
```

Or something to that extent?

Reply:
will say in the csv because thats what we check.. I have been assuming that if an scc is specified it should be in the csv so far ? I think thats the case with RBACs in general ?

Reply:
I _think_ you can include arbitrary RBAC in the manifests... I can never find the link to operator-sdk where they list the valid Kinds, though. All that said, We're only checking the CSV, so that's all fine. Just indicating custom vs. default?

Reply:
Since OLM doesn't apply custom SCC's and a user would need to, I think us just checking the CSV is fine, since the developer of the operator would have to mention their SCC in the CSV for the operator to know about it anyways. Us scanning the entire `/bundle` is overkill IMO.

Reply:
also added that if custom check operator docs as it may need to be defined..